### PR TITLE
Quick look - one or two tips.

### PR DESCRIPTION
**Pull request by Octavo8**
Hey hey Rob!

I just saw this now - I'm very impressed with what you've done so far, but it looks like you maybe gave up after putting in quite a bit of effort? I didn't spend huge amounts of time going through all of it, but I can see some clever ideas here.

I'm happy to spend time working with you through github to get it working if you want? It's a really neat idea.

### CODE REVIEW

File: TetrisBot/TetBrain.cs [hunk 1]
```diff
@@ -178,10 +175,10 @@ public void hitBottom(object sender, EventArgs e)
         private Move getHighestRatedMove(List<Move> moves)
         {
 
-            int maxRating = -100000000;
-            int maxMove = -1;
+            int maxRating = int.MinValue; //this guarentees the a lower bound.
+            int maxMove; // no need for this, = -1;
```
Comment (zing-rsa) [inline TetrisBot/TetBrain.cs:179]: Ahh nice, much cleaner!

File: TetrisBot/TetBrain.cs [hunk 2]
```diff
@@ -190,6 +187,8 @@ private Move getHighestRatedMove(List<Move> moves)
                     maxMove = i;
 
                     ties = new List<int>() { i };//NB find out what this does
+                    //ever time a move with a higher rating shows up, you create a new list of
+                    // ints, with the first element having the value of i
```
Comment (zing-rsa) [inline TetrisBot/TetBrain.cs:191]: Thanks, old comment I think

File: TetrisBot/TetBrain.cs [hunk 3]
```diff
@@ -216,13 +215,17 @@ public List<Move> getAllPossibleMoves()
...
                 List<int> originalPos = new List<int>();
 
                 for (int translations = -5; translations <= 5; translations++)
-                {
+                {//I don't get this loop. Why 10 translations? surely there are only 3 translations -
+                 //left, right and no move?  Also, why is this a nested loop? If we are looking at all possible
+                 //moves, their options are a rotation and a translation?
+
+                    //There's some interesting and cool code here. If you want help with it, let me know.
                     loadState(prevGameState);
```
Comment (zing-rsa) [inline TetrisBot/TetBrain.cs:229]: So, your game grid has an x-axis of 10 points right. As opposed to thinking about a move as either one* move right or left, I thought that 5 moves to the left can be considered as a single move(or potential position) to evaluate. Then 4 moves to the left, then 3 etc. all the way until 5 moves to the right. That way, the outcome of each position in the x-axis is evaluated from left to right. 
Does that make sense, or do you think there is a better way of doing this?

File: TetrisBot/TetBrain.cs [hunk 4]
```diff
@@ -1,13 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
-using System.Runtime.Serialization;
-using System.Runtime.Serialization.Formatters.Binary;
-using System.Text;
-using System.Threading.Tasks;
 using Tetris_Clone;
-using Newtonsoft.Json;
```
Comment (zing-rsa) [inline TetrisBot/TetBrain.cs:5]: Thanks, forgot to remove a bunch of usings from when I was trying to figure out how to deep clone.

**Review by zing-rsa — COMMENTED** [review]
Hey Dave, thanks for the input man! I haven't looked at this in a while yeah, managed to find myself a job, which has been keeping me pretty busy. But yes, I'd really appreciate it if you're keen to help finish it off. Would also be cool to sharpen up my git skills! Thanks for the comments, I have replied to some in a review.

I think I had some uncommitted changes on my pc repo that were lost when I had to reinstall windows, so there might be some stuff that is half done. Will have a look through to get a good understanding of what's what again.

File: TetrisBot/TetBrain.cs (hunk 3 above)
Comment (Octavo8) [reply TetrisBot/TetBrain.cs:229]: Ok, I think I see what you're getting at. Let me think about it a bit this weekend.

If you have a look at the fork on my page, I've added a wiki page with my ramblings, so you can see what I was thinking.

Also, don't forget to accept the pull request and merge :)

**Pull request merged**